File: app/src/main/java/com/github/kongpf8848/animation/views/StickerRecyclerView.java
package com.github.kongpf8848.animation.views;

import android.content.Context;
import android.util.AttributeSet;
import android.view.GestureDetector;
import android.view.MotionEvent;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.view.GestureDetectorCompat;
import androidx.recyclerview.widget.RecyclerView;

public class StickerRecyclerView extends RecyclerView {

    private GestureDetectorCompat gestureDetector;
    private OnItemClickListener onItemClickListener;
    private OnItemLongClickListener onItemLongClickListener;

    public interface OnItemClickListener {
        void onItemClick(View view, int position);
    }
    public interface OnItemLongClickListener {
        boolean onItemLongClick(View view, int position);
    }


    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }

    public void setOnItemLongClickListener(OnItemLongClickListener onItemLongClickListener) {
        this.onItemLongClickListener = onItemLongClickListener;
    }

    private class RecyclerListViewItemClickListener implements OnItemTouchListener {

        public RecyclerListViewItemClickListener(Context context) {
            gestureDetector=new GestureDetectorCompat(context, new GestureDetector.SimpleOnGestureListener(){
                @Override
                public boolean onSingleTapUp(MotionEvent e) {
                    View childView = findChildViewUnder(e.getX(),e.getY());
                    if(childView != null && onItemClickListener != null){
                        onItemClickListener.onItemClick(childView,getChildAdapterPosition(childView));
                        return true;
                    }
                    return false;

                }

                @Override
                public void onLongPress(MotionEvent e) {
                    View childView = findChildViewUnder(e.getX(),e.getY());
                    if(childView != null && onItemLongClickListener != null){
                        onItemLongClickListener.onItemLongClick(childView,getChildAdapterPosition(childView));
                    }

                }
            });
        }
        @Override
        public boolean onInterceptTouchEvent(@NonNull RecyclerView rv, @NonNull MotionEvent e) {
            if(gestureDetector.onTouchEvent(e)){
                return true;
            }else {
                return false;
            }
        }

        @Override
        public void onTouchEvent(@NonNull RecyclerView rv, @NonNull MotionEvent e) {

        }

        @Override
        public void onRequestDisallowInterceptTouchEvent(boolean disallowIntercept) {

        }
    }
    public StickerRecyclerView(@NonNull Context context) {
        this(context,null);
    }

    public StickerRecyclerView(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs,0);
    }

    public StickerRecyclerView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init(context);
    }

    private void init(Context context) {
        addOnItemTouchListener(new RecyclerListViewItemClickListener(getContext()));
    }
}

File: app/src/main/java/com/github/kongpf8848/animation/views/StickerRecyclerView.kt
package com.github.kongpf8848.animation.views

import android.content.Context
import android.util.AttributeSet
import android.view.GestureDetector.SimpleOnGestureListener
import android.view.MotionEvent
import android.view.View
import androidx.core.view.GestureDetectorCompat
import androidx.recyclerview.widget.RecyclerView

class StickerRecyclerView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0) : RecyclerView(context, attrs, defStyle) {

    private var gestureDetector: GestureDetectorCompat? = null
    private var onItemClickListener: OnItemClickListener? = null
    private var onItemLongClickListener: OnItemLongClickListener? = null

    interface OnItemClickListener {
        fun onItemClick(view: View, position: Int)
    }

    interface OnItemLongClickListener {
        fun onItemLongClick(view: View, position: Int)
    }

    fun setOnItemClickListener(onItemClickListener: OnItemClickListener?) {
        this.onItemClickListener = onItemClickListener
    }

    fun setOnItemLongClickListener(onItemLongClickListener: OnItemLongClickListener?) {
        this.onItemLongClickListener = onItemLongClickListener
    }

    init {
        addOnItemTouchListener(RecyclerListViewItemClickListener(getContext()))
    }

    private inner class RecyclerListViewItemClickListener(context: Context?) : OnItemTouchListener {

        init {
            gestureDetector = GestureDetectorCompat(context, object : SimpleOnGestureListener() {
                override fun onSingleTapUp(e: MotionEvent): Boolean {
                    val childView = findChildViewUnder(e.x, e.y)
                    if (childView != null && onItemClickListener != null) {
                        onItemClickListener!!.onItemClick(childView, getChildAdapterPosition(childView))
                        return true
                    }
                    return false
                }

                override fun onLongPress(e: MotionEvent) {
                    val childView = findChildViewUnder(e.x, e.y)
                    if (childView != null && onItemLongClickListener != null) {
                        onItemLongClickListener!!.onItemLongClick(childView, getChildAdapterPosition(childView))
                    }
                }
            })
        }

        override fun onInterceptTouchEvent(rv: RecyclerView, e: MotionEvent): Boolean {
            return gestureDetector!!.onTouchEvent(e)
        }
        override fun onTouchEvent(rv: RecyclerView, e: MotionEvent) {}
        override fun onRequestDisallowInterceptTouchEvent(disallowIntercept: Boolean) {}

    }


}